Signal/src/view controllers/MessageComposeTableViewController.m

```diff
@@ -543,16 +575,29 @@ - (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger
 - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
 {
     if (indexPath.section == MessageComposeTableViewControllerSectionInviteNonContactConversation) {
-        self.conversationForNonContactCell.textLabel.text = [NSString stringWithFormat:NSLocalizedString(@"NEW_CONVERSATION_FOR_NON_CONTACT_FORMAT",
+        if (indexPath.row < 0 ||
+            indexPath.row >= (NSInteger) self.searchPhoneNumberWithAccounts.count) {
+            OWSAssert(0);
+        }
+        NSString *phoneNumber = self.searchPhoneNumberWithAccounts[(NSUInteger) indexPath.row];
+        UITableViewCell *conversationForNonContactCell = [UITableViewCell new];
+        conversationForNonContactCell.textLabel.text = [NSString stringWithFormat:NSLocalizedString(@"NEW_CONVERSATION_FOR_NON_CONTACT_FORMAT",
                                                                                                          @"Text for button to start a new conversation with a non-contact"),
-                                                             self.searchPhoneNumber];
-        return self.conversationForNonContactCell;
+                                                             phoneNumber];
+        return conversationForNonContactCell;
     } else if (indexPath.section == MessageComposeTableViewControllerSectionInviteViaSMS) {
-        self.inviteViaSMSCell.textLabel.text = [NSString stringWithFormat:NSLocalizedString(@"SEND_INVITE_VIA_SMS_BUTTON_FORMAT",
+        if (indexPath.row < 0 ||
+            indexPath.row >= (NSInteger) self.searchPhoneNumbers.count) {
+            OWSAssert(0);
+        }
+        NSString *phoneNumber = self.searchPhoneNumbers[(NSUInteger) indexPath.row];
+        UITableViewCell *inviteViaSMSCell = [UITableViewCell new];
+        inviteViaSMSCell.textLabel.text = [NSString stringWithFormat:NSLocalizedString(@"SEND_INVITE_VIA_SMS_BUTTON_FORMAT",
                                                                                             @"Text for button to send a Signal invite via SMS. %@ is placeholder for the receipient's phone number."),
-                                                self.searchPhoneNumber];
-        return self.inviteViaSMSCell;
+                                                phoneNumber];
+        return inviteViaSMSCell;
     } else if (indexPath.section == MessageComposeTableViewControllerSectionInviteFlow) {
+        self.inviteCell.hidden = NO;
```

Comment:
I've modified `showEmptyBackgroundView` to show/hide all visible cells. It's much easier and safer to do that than trying to track references to all possibly visible cells as we were doing before.  However, we now need to ensure that all "reused" cells are not still hidden in this method.